tests/baselines/reference/noParameterReassignmentJSIIFE.types

```diff
@@ -0,0 +1,58 @@
+=== tests/cases/compiler/index.js ===
+"use strict";
+>"use strict" : "use strict"
+
+self.importScripts = (function (importScripts) {
+>self.importScripts = (function (importScripts) {  return function () {    setTimeout(function () {}, 0);    return importScripts.apply(this, arguments);  };})(importScripts) : (...args: string[]) => any
+>self.importScripts : (...urls: string[]) => void
+>self : Window & typeof globalThis
+>importScripts : (...urls: string[]) => void
+>(function (importScripts) {  return function () {    setTimeout(function () {}, 0);    return importScripts.apply(this, arguments);  };})(importScripts) : (...args: string[]) => any
+>(function (importScripts) {  return function () {    setTimeout(function () {}, 0);    return importScripts.apply(this, arguments);  };}) : (importScripts: (...urls: string[]) => void) => (...args: string[]) => any
+>function (importScripts) {  return function () {    setTimeout(function () {}, 0);    return importScripts.apply(this, arguments);  };} : (importScripts: (...urls: string[]) => void) => (...args: string[]) => any
+>importScripts : (...urls: string[]) => void
+
+  return function () {
+>function () {    setTimeout(function () {}, 0);    return importScripts.apply(this, arguments);  } : (...args: string[]) => any

```

Comment:
Previously, this line incorrectly had type `(..args: any[]) => any` because return type inference contextual parameter type fixing was being skipped! Now we pull the type from the `importScripts` signature as expected.